src/components/publicProfileSidebar/index.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  FollowButton,
+  FollowButtonWrapper,
+  ProfileAvatar,
+  SideBarInfo,
+  SideBarWrapper,
+  UserInfo,
+  UserName,
+  UserRating,
+} from 'components/publicProfileSidebar/styles.ts';
+import bgImg from 'assets/images/default_profile_img.png';
+import RatingStarIcon from 'assets/icons/iconRatingStar.tsx';
+import { PublicProfileSidebarType } from 'components/publicProfileSidebar/types.ts';
+import { useTranslation } from 'react-i18next';
+import {
+  useFollowUserMutation,
+  useGetIsFollowingQuery,
+} from 'redux/userApiSlice.ts';
+import {
+  isErrorWithMessage,
+  isFetchBaseQueryError,
+} from 'helpers/errorHandler.ts';
+import { useState } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const PublicProfileSidebar = ({
+  fullName,
+  rate,
+  userId,
+}: PublicProfileSidebarType) => {
+  const { t } = useTranslation();
+
+  const [error, setError] = useState<string>();
+
+  const { data: isFollowing, refetch } = useGetIsFollowingQuery(userId!);
+  const [followUser] = useFollowUserMutation();
+
+  const handleFollow = async () => {
+    try {
+      await followUser(userId!);
+      refetch();
+    } catch (err) {
+      if (isFetchBaseQueryError(err)) {
+        const errMsg = 'error' in err ? err.error : JSON.stringify(err.data);
```

Comment:
comparing with strings

Reply:
This is not a comparison. There is nothing wrong here.